neo/IO/Actors/PriorityMessageQueue.cs

```diff
@@ -34,9 +34,9 @@ public void Enqueue(IActorRef receiver, Envelope envelope)
         {
             Interlocked.Increment(ref idle);
             if (envelope.Message is Idle) return;
-            if (dropper(envelope.Message, high.Concat(low).Select(p => p.Message)))
-                return;
             ConcurrentQueue<Envelope> queue = priority_generator(envelope.Message) ? high : low;
+            if (dropper(envelope.Message, queue.Select(p => p.Message)))
```

Comment:
The dropper is a function to see if the message should be dropped. It is possible to write a function for the dropper that would want to check the message being enqueued using some criteria involving messages from either the high or low priorityqueue. 

Making this change only allows the dropper to to do checks on the message based on messages in the queue to which it will be placed rather than both queues. While this change would not do any harm currently, since the current dropper being used (`ProtocolHandler.ShallDrop`) only looks for duplicate messages for certain types to drop, future usage might want to check across queues.

That being said this design of passing an enumerable for all the messages to determine whether to drop is not very efficient. Each time a message is enqueued it potentially examines all messages. We might want to consider a design that keeps more information about enqueued messages so it could immediately tell whether a message was a duplicate without having to do an O(n) operation.